GetJobber/atlantis
# docs: Link Checking + E2E + Correct Storybook Links for All Components

## Motivations

Most of the "Storybook" links in the right-hand panel were broken prior to this PR and I wanted to fix those.

But since those are just links based on strings in files which are relatively brittle to change, I wanted to introduce a regression test to make sure that future changes don't disrupt all of the links being functional.

So I've introduced a Playwright based test suite that is entirely optional to run. It's not currently part of any automation, but you can run the test suite whenever you want to find any broken links.

We could certainly add it to our automation in the future, but it's a little expensive to run. It takes about 4 minutes currently for a successful run. It runs each test 3 times, once for Chrome, Firefox, and Safari. In Desktop mode. We can optionally run more devices in the future if we find this useful.

## Changes

1. Fixed all the Storybook links in the side panel.
2. Added a Playwright based regression test to ensure all of our pages are rendering properly (with the exception of the design pages, will update in a future ticket)
3. Fixed rendering of the SideDrawer component (found by the tests!)
4. Broke out the base list of components in generateDocs into a shared .mjs file so I didn't have to duplicate the list of components.
5. Introduced a VITE_STORYBOOK_URL env var that swaps the link depending on which environment you're running in.



## Testing

1. Pull this branch down.
1. Run `npm ci`. 
1. Run `npx playwright install` This is going to take a minute to pull down the headless browsers.
1. Start the docs site
1. Start storybook
1. `npm run e2e` when you're in `packages/site`
1. The test suite should take about 5 minutes to run. It'll print out the component names as it runs through to the console. There is an HTML file thats generated at the end that you can look at if you're interested.

If the test suite does not run successfully, let me know! I might have to amend these instructions with more details.

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)
